src/main/java/io/jsonwebtoken/BadAudienceJwtException.java

```diff
@@ -0,0 +1,11 @@
+package io.jsonwebtoken;
+
+public class BadAudienceJwtException extends ClaimJwtException {
```

Comment:
I thought about `UnexpectedClaimException` but avoided it - I think it can too easily confuse people because of claims that exist, and are legal, but aren't specifically listed when building the parser.

For example, I don't 'expect' an IAT field, but it is there.  Unexpected might be seen as "a claim exists that I didn't expect to see" as opposed to "the value for a required claim is not the correct value"